server/src/Irony.cpp

```diff
@@ -433,6 +433,13 @@ void Irony::candidates() const {
         typedTextSet = true;
       }
     }
+    if (prefix != "" && prefix != "*") {
+      auto res = std::mismatch(prefix.begin(), prefix.end(), typedtext.begin());
```

Comment:
Would be nice to make a function for that, something like `bool startsWith(const std::string &str, const std::string &prefix)`.

Also, could you make this check this earlier? In order to avoid the unnecessary processing that is going on to make the annotation, brief and all. In the if-block of line 429 for example: `if (chunkKind == CXCompletionChunk_TypedText && !typedTextSet)`.